src/main/java/il/co/codeguru/corewars8086/gui/effects/VisualEffectManager.java

```diff
@@ -0,0 +1,80 @@
+package il.co.codeguru.corewars8086.gui.effects;
+
+import java.awt.Graphics;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Manages all active visual effects.
+ * Handles updating and rendering effects, and removes completed ones.
+ */
+public class VisualEffectManager {
+    private final List<VisualEffect> effects;
+    
+    public VisualEffectManager() {
+        this.effects = new ArrayList<>();
```

Comment:
Seeing as all uses of this list are synchronized, I suggest wrapping it in `Collections.synchronizedList`